core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/seo/components/dot-device-selector-seo/dot-device-selector-seo.component.spec.ts

```diff
@@ -0,0 +1,147 @@
+})
+class TestHostComponent {}
+
+describe('DotDeviceSelectorSeoComponent', () => {
+    let fixtureHost: ComponentFixture<TestHostComponent>;
+    let deHost: DebugElement;
+    let component: DotDeviceSelectorSeoComponent;
+    let de: DebugElement;
+
+    const messageServiceMock = new MockDotMessageService({
+        'editpage.device.selector.title': 'Devices',
+        'editpage.device.selector.media.tile': 'Social Media Tiles',
+        'editpage.device.selector.search.engine': 'Search Engine Results Pages',
+        'editpage.device.selector.new.tab': 'Open in New Tab',
+        'editpage.device.selector.mobile.portrait': 'Mobile Portrait',
+        'editpage.device.selector.mobile.landscape': 'Mobile Landscape',
+        'editpage.device.selector.hd.monitor': 'HD Monitor',
+        'editpage.device.selector.4k.monitor': '4K Monitor',
+        'editpage.device.selector.tablet.portrait': 'Tablet Portrait',
+        'editpage.device.selector.tablet.landscape': 'Tablet Landscape'
+    });
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            declarations: [TestHostComponent],
+            imports: [
+                DotDeviceSelectorSeoComponent,
+                HttpClientTestingModule,
+                OverlayPanelModule,
+                BrowserAnimationsModule
+            ],
+            providers: [
+                {
+                    provide: DotDevicesService,
+                    useClass: DotDevicesServiceMock
+                },
+                {
+                    provide: DotMessageService,
+                    useValue: messageServiceMock
+                },
+                {
+                    provide: CoreWebService,
+                    useClass: CoreWebServiceMock
+                }
+            ],
+            schemas: [CUSTOM_ELEMENTS_SCHEMA]
+        }).compileComponents();
+    });
+
+    beforeEach(() => {
+        fixtureHost = TestBed.createComponent(TestHostComponent);
+        deHost = fixtureHost.debugElement;
+        de = deHost.query(By.css('dot-device-selector-seo'));
+        component = de.componentInstance;
+        TestBed.inject(DotDevicesService);
+        spyOn(component, 'getOptions').and.returnValue(of(mockDotDevices));
+    });
+
+    it('should emit selected device on change', async () => {
+        spyOn(component.selected, 'emit');
+        spyOn(component, 'change');
+
+        const selectorButton: DebugElement = de.query(
+            By.css('[data-testId="device-selector-button"]')
+        );
+
+        selectorButton.nativeElement.click();
+
+        await fixtureHost.whenStable();
+        fixtureHost.detectChanges();
+
+        const selectorOptions = fixtureHost.debugElement.queryAll(
+            By.css('[data-testId="device-selector-option"] > .device-list__button')
+        );
+
+        selectorOptions[0].nativeElement.click();
+
+        fixtureHost.detectChanges();
+
+        expect(component.change).toHaveBeenCalled();
```

Comment:
You should assert the emit, because again I remove the emit from the body in the change method and the test will be green but the functionality will be broken.